unity/Assets/Scripts/PhysicsRemoteFPSAgentController.cs

```diff
@@ -5910,8 +5876,8 @@ bool markActionFinished
                     // Debug.Log(Vector3.Distance(tmp, transform.position));
                     if (Vector3.Distance(tmp, transform.position) < maxDistance) {
                         // if this particular point is in view...
-                        if (CheckIfVisibilityPointInViewport(sop, point, m_Camera, false) ||
-                            CheckIfVisibilityPointInViewport(sop, point, m_Camera, true)) {
+                        if ((CheckIfVisibilityPointInViewport(sop, point, m_Camera, false) |
+                            CheckIfVisibilityPointInViewport(sop, point, m_Camera, true)).visible) {
```

Comment:
This would be better as 
```
                        if ((CheckIfVisibilityPointInViewport(sop, point, m_Camera, false).visible ||
                            CheckIfVisibilityPointInViewport(sop, point, m_Camera, true).visible)) {
```
as using `||` allows C# to skip running the second visibility check if the first one is true.

Reply:
updated!